src/quantile.js

```diff
@@ -1,6 +1,19 @@
 import number from "./number.js";
+import identity from "./identity.js";
 
 export default function quantile(values, p, valueof = number) {
+  values = Array.from(values)
+    .map((d, i) => valueof(d, i, values))
+    .filter(v => !isNaN(v))
+    .sort((a, b) => a - b);
+  var n = values.length;
+  if (!n) return;
+  if ((p = +p) <= 0 || n < 2) return +values[0];
+  if (p >= 1) return +values.pop();
+  return quantileSorted(values, p, identity);
```

Comment:
I think I would simplify and not pass `identity` here, since the default `number` will do just as well, and will only be called at most twice.